### PR TITLE
Install jupyter_packaging in tutorial

### LINKED ISSUE #9174: Issues with 3.0.0-rc5 tutorial
## Description

Attempting to go through the `jupyterlab_apod` demo with jupyterlab 3.0.0-rc5

## Reproduce

Followed directions. I think the point where my setup differs is that I was trying to install into `~/.local`

## Issues

1. jupyter_packaging was not mentioned until later, but is needed much earlier.  Suggest adding `pip install jupyter_packaging` with jupyterlab
2. `pip install -ve .` does not work with `--user`; I had to use `--prefix=~/.local`
3. Even then, there is an error in where to put /share/ items.

```
$ pip install -ve . --prefix=/home/dblank/.local
...
Creating /home/dblank/.local/lib/python3.8/site-packages/jupyterlab-apod.egg-link (link to .)
Adding jupyterlab-apod 0.1.0 to easy-install.pth file

Installed /home/dblank/jupyterlab_apod
creating /usr/share/jupyter
error: could not create '/usr/share/jupyter': Permission denied
```

### PR DESCRIPTION
**Pull request by jasongrout**
## References

Fixes #9174

## User-facing changes

Adds the step to install jupyter-packaging in the tutorial instructions

### CODE REVIEW

File: docs/source/developer/extension_tutorial.rst [hunk 1]
```diff
@@ -162,6 +162,7 @@ dependencies and install the extension into the JupyterLab environment.
 
 .. code:: bash
 
+    pip install jupyter_packaging
```
Comment (jtpio) [inline docs/source/developer/extension_tutorial.rst:165]: An alternative could be to have it as part of the `conda create` command above?

Comment (jasongrout) [reply docs/source/developer/extension_tutorial.rst:165]: Yep. Do you prefer it there? I thought it fit here as an "initial project dependency", but I could also see it up in the conda create above. I don't have a strong preference either way.

Comment (jasongrout) [reply docs/source/developer/extension_tutorial.rst]: I moved it up to the conda environment creation.

**Comment by ianhi** [comment]
Maybe also in the guide for updating extensions? It seems you need both `cookiecutter` and `jupyter_packaging` to run the upgrade script

**Review by jtpio — APPROVED** [review]
Thanks!

**Comment by jtpio** [comment]
> Maybe also in the guide for updating extensions? It seems you need both cookiecutter and jupyter_packaging to run the upgrade script

Thanks @ianhi, sounds good :+1: 

Just opened https://github.com/jupyterlab/jupyterlab/pull/9199 if you want to have a look.

**Pull request merged**